### PR TITLE
HDDS-5144. Create github check to alert when dependency tree is changed

### PR DESCRIPTION
**Pull request by elek**
JIRA: https://issues.apache.org/jira/browse/HDDS-5144

## What changes were proposed in this pull request?

`hadoop-ozone/dist/src/main/license/bin/LICENSE.txt` and `hadoop-ozone/dist/src/main/license/bin/LICENSE.txt` should be modified when the transitive dependencies are changed.

However, it's very easy to miss any dependency change when we bump versions. It's possible that we change only versions but we also introduce new jar files via transitive dependencies.

This happened (for example) in #2139 and #2065 

To avoid any mistake I suggest the same approach what we use the protobuf compatibility check:

1. commit the current list of the jar files to the repository
2. create a check which compares the actual value with the committed one

If new dependency is introduced, the LICENSE file can be updated and the committed jar list also can be updated.

## How was this patch tested?

Tested on my fork. 

Example of failing builds:

https://github.com/elek/ozone/runs/2407951331?check_suite_focus=true

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @elek for working on this check.  The list of jars is not up-to-date wrt. `master`, so the new check fails in this PR.

File: .github/workflows/post-commit.yml [hunk 1]
```diff
@@ -112,6 +112,33 @@ jobs:
           rm -rf ~/.m2/repository/org/apache/hadoop/hadoop-hdds*
           rm -rf ~/.m2/repository/org/apache/hadoop/hadoop-ozone*
         if: always()
+  dependency:
+    needs: compile
+    runs-on: ubuntu-18.04
+    timeout-minutes: 300
```
[the remark this answers is not in the document]
Comment (elek) [reply .github/workflows/post-commit.yml:118]: yep. Let me change it to 5 minutes.

**Review by adoroszlai — APPROVED** [review]
Thanks @elek for updating the patch.

**Comment by elek** [comment]
Thanks the review @adoroszlai. I am merging it now.

**Pull request merged**